azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go

```diff
@@ -0,0 +1,159 @@
+package apimanagement
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2018-01-01/apimanagement"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmApiManagementIdentityProviderGoogle() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApiManagementIdentityProviderGoogleCreateUpdate,
+		Read:   resourceArmApiManagementIdentityProviderGoogleRead,
+		Update: resourceArmApiManagementIdentityProviderGoogleCreateUpdate,
+		Delete: resourceArmApiManagementIdentityProviderGoogleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"api_management_name": azure.SchemaApiManagementName(),
+
+			"client_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"client_secret": {
```

Comment:
Could we change this to match what google calls this: 
```suggestion
			"api_ey": {
```

Reply:
I believe both Azure and Google actually refer to the value as client secret. I'll update the documentation to reflect this.

Reply:
If you look at the SDK docs (and presumably the API) it states:
```
// ClientSecret - Client secret of the Application in external Identity Provider, used to authenticate login request. For example, it is App Secret for Facebook login, API Key for Google login, Public Key for Microsoft.
```

Is this inaccurate and client secret is what you would look for on the google side?

Reply:
Yep, it is called client secret as well on the Google side.

![google](https://user-images.githubusercontent.com/39076898/71647931-d6f6d680-2cc2-11ea-915b-d4421dd79cba.jpg)

Reply:
perfect, thanks for that!